### ISSUE TITLE
Substitution/concatenation of token names

### ISSUE THREAD

**Issue by oeed** [Emoun/duplicate#52]
### Short Description:
I was hoping to have a way to substitute a value in to the name of a token. For example:

```rust
#[duplicate_item(
  int_type  max_value;
  [ u8 ]    [ 255 ];
  [ u16 ]   [ 65_535 ];
  [ u32 ]   [ 4_294_967_295 ];
)]
fn is_[int_type]_max(&self) -> bool {
  *self == max_value
}

// Generates...
fn is_u8_max(&self) -> bool {
  *self == 255
}

fn is_u16_max(&self) -> bool {
  *self == 65_535
}
// etc.
```

I've read through the docs and can't seem to find a way to do so without restoring to using something like `paste!`. Is that possible?

**Comment by oeed** [comment]
Yeah that's totally fair, no specific issues with using `paste!` - although as I haven't had to use it before I had to Google around for a while before discovering how to do so. Might just be a matter of an example in the docs.

**Comment by Emoun** [comment]
I've thought a little bit about this since yesterday.

The `module_disambiguation` is kind of like this feature: automatically renaming an item based on the substitutions.
I have clearly previously found enough value in it to add it.
Maybe there is value in a more general feature, perhaps called `item_disambiguation`, that works the same as `module_disambiguation` but for any kind of item. 
This would cover the above example:

```rust
#[duplicate_item(
  int_type  max_value;
  [ u8 ]    [ 255 ];
  [ u16 ]   [ 65_535 ];
  [ u32 ]   [ 4_294_967_295 ];
)]
fn is_max(&self) -> bool {
  *self == max_value
}

// Generates...
fn is_max_u8(&self) -> bool {
  *self == 255
}

fn is_max_u16(&self) -> bool {
  *self == 65_535
}
// etc.
```

Pros:

* An extension to an existing feature.
* Limited in scope.
* Conceivably useful for many users. [Here is another example](https://stackoverflow.com/questions/74512752/generating-functions-over-a-range-of-numbers/74517361#74517361).

Cons:

* Significant added complexity, as it requires the crate to analyze the code to check if the body is an item and find the name.
* It requires us to add the exact concatenation rules to the stability guarantees, such that people can call the resulting functions or structs without worrying the name might change in the future.

Question:
@oeed how would you feel about not being able to control exactly where in the name the substitution is added? E.g. that it will always be at the end: `is_max_u8`, `is_max_u16` etc.

If this was available to you, would you have used it?
